modules/markup/markdown/meta.go

```diff
@@ -25,20 +25,22 @@ func isYAMLSeparator(line string) bool {
 // and returns the frontmatter metadata separated from the markdown content
 func ExtractMetadata(contents string, out interface{}) (string, error) {
 	var front, body []string
-	var seps int
 	lines := strings.Split(contents, "\n")
 	for idx, line := range lines {
```

Comment:
It's better to use `Scanner`.
```go
scanner := bufio.NewScanner(strings.NewReader(contents))
var idx = 0 
for scanner.Scan() {
    line := scanner.Text()
    ...
    idx++
}
```

Reply:
What's the advantage in this case?
We still need to scan the whole file so we can return body, and we've already checked file size prior to calling this func.

Reply:
Less CPU and memory usage. But since it should be a small file, it's not necessary.